ex5/ex5.c

```diff
@@ -16,6 +17,20 @@ char* msg3 = "hello world #3";
 int main()
 {
     // Your code here
-    
+    int fds[2];
+    char buffer[128];
+
+    int child = fork();
+
+    if (child < 0) {
+      printf("Cannot fork a child\n");
+    } else if (child == 0) {
+      write(fds[1], msg1, strlen(msg1));
+    } else {
+      wait(NULL);
+      read(fds[0], buffer, 128);
```

Comment:
Likewise, you'll have to change the parent's code too to read all three of the messages off of the pipe.